sympy/series/approximants.py

```diff
@@ -101,3 +108,282 @@ def approximants(l, X=Symbol('x'), simplify=False):
+    ) -> tuple[Poly, Poly] | None:
+    """
+    `[m/n]` pade approximant of `f` around `x=0`. Computed using the extended euclidean algorithm.
+
+    Description
+    ===========
+
+    For a polynomial `f` and integers `m` and `n`, the `[m/n]` pade approximant
+    of `f` is a rational function `p(x)/q(x)`, where `p(x)` and `q(x)` are
+    polynomials of degree at most `m` and `n` respectively. The pade
+    approximation, when it exists, is such that the taylor series of `p(x)/q(x)`
+    around `x=0`matches `f` up to order `(m + n)` in `x`.
+
+    Parameters
+    ==========
+
+    f : Poly
+        The polynomial to approximate
+    x : Expr
+        The variable of the function
+    m : int
+        The degree of the numerator polynomial
+    n : int | None
+        The degree of the denominator polynomial. If None, `n = m`
+
+    Returns
+    =======
+
+    pade approximant : tuple[Poly, Poly]
+        returns p(x), q(x), where p(x)/q(x) is the pade approximant
+
+    Examples
+    ========
+
+    >>> import sympy as sp
+    >>> from sympy.series.approximants import pade_approximant_gcdex
+    >>> x = sp.symbols('x')
+
+    >>> exp_taylor_poly = sp.Poly(x**4/24 + x**3/6 + x**2/2 + x + 1, x, domain='QQ')
+    >>> num_exp, denom_exp = pade_approximant_gcdex(exp_taylor_poly, x, 2)
+    >>> print(num_exp, denom_exp)
+    Poly(1/4*x**2 + 3/2*x + 3, x, domain='QQ') Poly(1/4*x**2 - 3/2*x + 3, x, domain='QQ')
+
+    The numerators and denominators of an `[m/n]` pade approximant do not
+    necessarily have order `m` and `n` respectively.
+
+    >>> sin_taylor_poly = sp.Poly(x**5/120 - x**3/6 + x, x, domain='QQ')
+    >>> num_sin, denom_sin = pade_approximant_gcdex(sin_taylor_poly, x, 1, 3)
+    >>> print(num_sin, denom_sin)
+    Poly(36*x, x, domain='QQ') Poly(6*x**2 + 36, x, domain='QQ')
+
+    The `[m/0]` pade approximant is the same a truncating to `m`th order
+
+    >>> poly = sp.Poly(4*x**4 + 3*x**3 + 2*x**2 + x + 1, x, domain='QQ')
+    >>> num_cos, denom_cos = pade_approximant_gcdex(poly, x, 3, 0)
+    >>> print(num_cos, denom_cos)
+    Poly(3*x**3 + 2*x**2 + x + 1, x, domain='ZZ') Poly(1, x, domain='ZZ')
+
+    The pade approximant does not always exist, for example, the `[1/1]` approximant for `cos(x)`
+
+    >>> cos_taylor_poly = sp.Poly(1 - x**2/2, x, domain='QQ')
+    >>> num_cos11, denom_cos11 = pade_approximant_gcdex(cos_taylor_poly, x, 1)
+    >>> print(num_cos11, denom_cos11)
+    Poly(2*x, x, domain='QQ') Poly(2*x, x, domain='QQ')
+
+    `num_cos11/denom_cos11 == 1`, and `1` does not match the taylor expansion of `cos(x)`
+    to second order in `x`.
+
+    See also
+    ========
+
+    sympy.series.approximants.approximants
+    sympy.series.approximants.pade_approximant
+    sympy.series.approximants.pade_approximants
+    """
+    if n is None:
+        n = m
+
+    if m < 0 or n < 0:
+        raise ValueError("m and n must be non-negative integers")
+
+    min_degree = min(f.as_dict().keys())[0]
```

Comment:
There is `f.EM()[0]` for this, not that it is implemented particularly efficiently.